### PR TITLE
fixes issue #87

### PR DESCRIPTION
**Pull request by dopefishh**
It adds the search options in the hash. Probably we should include a
possibility to add page numbers as well. Moreover we should add cloo.gl
integration.

It also removes some old debug console.log's

Maybe we should even add cloo.gl integration before merging...

### CODE REVIEW

**Comment by camilstaps** [comment]
I thought the idea was to not put these things in the hash by default but accept hashes that contain this information. That way, the location bar is kept clean.

File: frontend/api.js [hunk 1]
```diff
@@ -417,9 +415,15 @@ function getResults(str, libs, include_builtins, include_core, page) {
 
 	xmlHttp.open('GET', url, true); // true for asynchronous
 	xmlHttp.send(null);
-	if (document.location.hash.substring(1) != encodeURIComponent(str)) {
+	if (true || document.location.hash.substring(1) != encodeURIComponent(str)) {
```
Comment (camilstaps) [inline frontend/api.js]: Why this change? We should not set `refresh_on_hash` to false when the request has not been changed, because this will cause unexpected page reloads.

Comment (dopefishh) [reply frontend/api.js]: true, I'll change this. the comparison needs to be improved though. The string only doesn't suffice anymore

**Comment by dopefishh** [comment]
Only advanced settings are put in the hash

**Comment by dopefishh** [comment]
Where shall I put the cloo.gl reference

**Comment by dopefishh** [comment]
Share button is added. Can you apply your UI/UX magic @ErinvanderVeen?

**Comment by camilstaps** [comment]
Okay great, I'm happy with the hash as it is now.

I would suggest the following UI changes:

- Only show the share option when a request has been made
- I don't think a `.tooltip` is needed for this. We could just open a javascript alert on error, and a prompt with the URL on success. In most browsers this selects the text directly so it's easy to copy. Like this, sharing the query is only one click away.

**Comment by dopefishh** [comment]
no alerts please, sane browser configuration block those.
Maybe we can move the functionality of the share button in the tooltip-opener itself

**Comment by camilstaps** [comment]
Browsers don't block alerts, they block pop-up windows. But I agree that it's not ideal. Moving the text to the link would be good, then we just need to find a way to give the link to the user.
